artemis-server/src/main/java/org/apache/activemq/artemis/core/paging/impl/PagingStoreImpl.java

```diff
@@ -228,11 +229,34 @@ private void configureSizeMetric() {
       size.setMax(maxSize, maxSize, maxMessages, maxMessages);
    }
 
+   private void validateNewSettings(final AddressSettings addressSettings) throws ActiveMQException {
+
+      Long newPageLimitBytes = addressSettings.getPageLimitBytes();
+
+      if (newPageLimitBytes != null && newPageLimitBytes.longValue() < 0) {
+         newPageLimitBytes = null;
+      }
+      int newPageSize = storageManager.getAllowedPageSize(addressSettings.getPageSizeBytes());
+      if (newPageLimitBytes != null && newPageSize > 0) {
+         long newEstimatedMaxPages = newPageLimitBytes / newPageSize;
+         if (this.numberOfPages > newEstimatedMaxPages) {
+            throw ActiveMQMessageBundle.BUNDLE.invalidPageSettings(this.storeName.toString(), addressSettings, "estimated max pages " + newEstimatedMaxPages + " is less than current number of pages " + this.numberOfPages);
```

Comment:
You should log.warn here, from the Logger, and return false, 

on the applySettings you should then if (!valideSettings) return;


because here you are throwing an exception just to log.warn... It will be more complicated than it needs to be here.

Reply:
Make sense. Thanks!